pennylane/transforms/optimization/merge_rotations.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Transform for merging adjacent rotations of the same type in a quantum circuit."""
+
+from pennylane import apply
+from pennylane.transforms import qfunc_transform
+from pennylane.math import allclose, stack, cast_like, zeros
+
+from .optimization_utils import _find_next_gate, _fuse_rot_angles
+
+
+@qfunc_transform
+def merge_rotations(tape):
```

Comment:
is there any utility in allowing only specific rotation gates be selected?

e.g.,
```python
@qml.qnode(dev)
@merge_rotations(tape, ops={"RX", "RY", "CRZ"})
def qfunc(params):
```
(can't think of a better argument name 🤔 )

Reply:
Yes this is a good idea; another suggestion from Nathan was to add an accuracy threshold for the angle cancellation. 

I'd been wondering whether having a mix of parametrized and non-parametrized functions would make it more difficult to write the `compile` function though? Or could the pipeline be specified and applied like this:

```python
pipeline = [cancel_inverses, merge_rotations(ops=["RX", "CRZ"])]

for transform in pipeline:
    qfunc = transform(qfunc)

Reply:
> another suggestion from Nathan was to add an accuracy threshold for the angle cancellation.

Good idea!

> Or could the pipeline be specified and applied like this:

Omg I think this would work! I've often despaired at `transform(params)(func)(args)` being less-than-intuitive, but this might be an advantage!

Reply:
Okay amazing, I will update this then!